File: android/src/main/kotlin/com/example/pdf/resources/DocumentRepository.kt
package com.example.pdf.resources

import android.annotation.TargetApi
import android.graphics.pdf.PdfRenderer
import android.os.Build
import android.os.ParcelFileDescriptor
import com.example.pdf.document.Document
import com.example.pdf.utils.randomID

@TargetApi(Build.VERSION_CODES.LOLLIPOP)
class DocumentRepository : Repository<Document>() {
    /**
     * Register document in repository
     * @returns document id
     */
    fun register(getPair: Pair<ParcelFileDescriptor, PdfRenderer>?): Document {
        val id = randomID
        val (fileDescriptor, renderer) = getPair!!
        val document = Document(id, renderer, fileDescriptor)
        set(id, document)
        return document
    }

    public override fun close(id: String) {
        get(id).close()
        super.close(id)
    }
}

File: ios/Classes/resources/DocumentRepository.swift
//
//  DocumentRepository.swift
//  pdf
//
//  Created by 林炳午 on 2024/3/8.
//

import Foundation
class DocumentRepository : Repository<Document> {
    func register(renderer: CGPDFDocument) -> Document {
        let id = NSUUID().uuidString
        let page = Document(id: id, renderer: renderer)
        set(id: id, item: page)
        return page
    }
}
